src/sort.rs

```diff
@@ -80,16 +122,17 @@ pub fn sort_toml(
                     // The root table is always index 0 which we ignore so add 1
                     first_table = Some(idx + 1);
                 }
-                let headings = heading_order.entry((idx, head.to_string())).or_default();
+                let key = item_key.to_string();
+                let headings = heading_order.entry((idx, key.clone())).or_default();
                 // Push a `Heading::Complete` here incase the tables are ordered
                 // [heading.segs]
                 // [heading]
                 // It will just be ignored if not the case
-                headings.push(Heading::Complete(vec![head.to_string()]));
+                headings.push(Heading::Complete(vec![key]));
 
                 gather_headings(table, headings, 1);
                 headings.sort();
-                sort_table(table, group);
+                sort_table(table, group, &target_tables);
```

Comment:
This instance of `target_tables` will always be empty, right? I think that confirms that we should have two different functions here:

- `sort_table`
- `sort_nested_table`

Perhaps?

Reply:
done